### ISSUE TITLE
Cannot build Docker image

### ISSUE THREAD

**Issue by sawyercade** [enigmampc/catalyst#366]
# Environment

* Operating System: Darwin Kernel Version 17.3.0
* Python Version: Python 3.6.4 :: Anaconda, Inc.
* Python Bitness: 64
* How did you install Catalyst: `git clone`, `docker build`
* Python packages: N/A

# Description of Issue

I'm attempting to build a docker image using the root Dockerfile. It throws the following exception when I run `docker build -t enigmampc/catalyst .` from the root project directory, on `Step 14/16 : RUN pip install -e .`

```
Successfully built Logbook cyordereddict bottleneck networkx bcolz toolz Mako sqlalchemy alembic intervaltree lru-dict empyrical redo requests-ftp wrapt python-editor cytoolz attrdict parsimonious
Exception:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2862, in _dep_map
    return self.__dep_map
  File "/usr/local/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2669, in __getattr__
    raise AttributeError(attr)
AttributeError: _DistInfoDistribution__dep_map

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2853, in _parsed_pkg_info
    return self._pkg_info
  File "/usr/local/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2669, in __getattr__
    raise AttributeError(attr)
AttributeError: _pkg_info

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/pip/_internal/basecommand.py", line 228, in main
    status = self.run(options, args)
  File "/usr/local/lib/python3.5/site-packages/pip/_internal/commands/install.py", line 318, in run
    self._warn_about_conflicts(to_install)
  File "/usr/local/lib/python3.5/site-packages/pip/_internal/commands/install.py", line 442, in _warn_about_conflicts
    package_set, _dep_info = check_install_conflicts(to_install)
  File "/usr/local/lib/python3.5/site-packages/pip/_internal/operations/check.py", line 89, in check_install_conflicts
    state = create_package_set_from_installed()
  File "/usr/local/lib/python3.5/site-packages/pip/_internal/operations/check.py", line 39, in create_package_set_from_installed
    retval[name] = PackageDetails(dist.version, dist.requires())
  File "/usr/local/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2613, in requires
    dm = self._dep_map
  File "/usr/local/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2864, in _dep_map
    self.__dep_map = self._compute_dependencies()
  File "/usr/local/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2873, in _compute_dependencies
    for req in self._parsed_pkg_info.get_all('Requires-Dist') or []:
  File "/usr/local/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 2855, in _parsed_pkg_info
    metadata = self.get_metadata(self.PKG_INFO)
  File "/usr/local/lib/python3.5/site-packages/pip/_vendor/pkg_resources/__init__.py", line 1405, in get_metadata
    return value.decode('utf-8') if six.PY3 else value
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xed in position 0: invalid continuation byte
The command '/bin/sh -c pip install -e .' returned a non-zero code: 2
```

## Reproduction Steps

1. Clone the catalyst repo
2. `docker build -t enigmampc/catalyst .`

Is this Dockerfile intended to be maintained?

**Comment by sawyercade** [comment]
Able to build and use the image now, thanks!

**Comment by pascalandy** [comment]
I just build and try the docker image from `develop`. 

It's working but not over http`s`

## terminal

```
➜  Documents docker logs -f catalyst
Writing default config to: /root/.jupyter/jupyter_notebook_config.py
Generating a 2048 bit RSA private key
..........+++
.........................................................+++
writing new private key to '/root/.jupyter/jupyter.key'
-----
[I 02:17:26.302 NotebookApp] Writing notebook server cookie secret to /root/.local/share/jupyter/runtime/notebook_cookie_secret
[I 02:17:26.586 NotebookApp] Serving notebooks from local directory: /projects
[I 02:17:26.586 NotebookApp] 0 active kernels
[I 02:17:26.586 NotebookApp] The Jupyter Notebook is running at:
[I 02:17:26.586 NotebookApp] https://6f158dcd4256:8888/
[I 02:17:26.586 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[W 02:22:45.666 NotebookApp] SSL Error on 9 ('172.17.0.1', 33902): [SSL: HTTP_REQUEST] http request (_ssl.c:833)
[W 02:22:45.667 NotebookApp] SSL Error on 10 ('172.17.0.1', 33904): [SSL: HTTP_REQUEST] http request (_ssl.c:833)
[W 02:22:45.676 NotebookApp] SSL Error on 11 ('172.17.0.1', 33906): [SSL: HTTP_REQUEST] http request (_ssl.c:833)
[W 02:22:45.681 NotebookApp] SSL Error on 9 ('172.17.0.1', 33908): [SSL: HTTP_REQUEST] http request (_ssl.c:833)
[W 02:22:45.912 NotebookApp] SSL Error on 9 ('172.17.0.1', 33910): [SSL: HTTP_REQUEST] http request (_ssl.c:833)
[W 02:22:45.917 NotebookApp] SSL Error on 10 ('172.17.0.1', 33912): [SSL: HTTP_REQUEST] http request (_ssl.c:833)
[W 02:22:45.936 NotebookApp] SSL Error on 9 ('172.17.0.1', 33914): [SSL: HTTP_REQUEST] http request (_ssl.c:833)
[W 02:22:45.945 NotebookApp] SSL Error on 9 ('172.17.0.1', 33916): [SSL: HTTP_REQUEST] http request (_ssl.c:833)
[W 02:22:50.972 NotebookApp] SSL Error on 10 ('172.17.0.1', 33920): [SSL: HTTP_REQUEST] http request (_ssl.c:833)
[W 02:22:50.975 NotebookApp] SSL Error on 9 ('172.17.0.1', 33918): [SSL: HTTP_REQUEST] http request (_ssl.c:833)
[W 02:22:50.980 NotebookApp] SSL Error on 9 ('172.17.0.1', 33922): [SSL: HTTP_REQUEST] http request (_ssl.c:833)
[W 02:22:50.983 NotebookApp] SSL Error on 9 ('172.17.0.1', 33924): [SSL: HTTP_REQUEST] http request (_ssl.c:833)
[I 02:23:12.752 NotebookApp] 302 GET / (172.17.0.1) 0.83ms
[I 02:23:12.759 NotebookApp] 302 GET /tree? (172.17.0.1) 0.90ms
[I 02:23:25.734 NotebookApp] 302 POST /login?next=%2Ftree%3F (172.17.0.1) 1.71ms
[I 02:23:32.065 NotebookApp] 302 GET /login?next=%2Ftree%3F (172.17.0.1) 1.37ms
[W 02:27:24.939 NotebookApp] 401 POST /login?next=%2F (172.17.0.1) 2.70ms referer=https://127.0.0.1:8888/login
[I 02:28:01.221 NotebookApp] 302 POST /login?next=%2F (172.17.0.1) 2.40ms
[I 02:28:01.232 NotebookApp] 302 GET / (172.17.0.1) 1.09ms
```

**Comment by pascalandy** [comment]
Hi again,

The docker image size is 2.06GB (uncompressed). I feel a multistage build would make more sense. 
I'll be glad to help on that.

Just buzz me!
https://twitter.com/askpascalandy

**Comment by lacabra** [comment]
HI @pascalandy,

Thanks for the follow up in reporting. One thought before I delve any further, it gets harder to track issues if we bundle them together in one single thread, so we very much prefer opening a new issue for any new problem that you encounter.

I can possibly look into the https issue in a couple of weeks. This is currently a low priority and we're working on meeting a competing deadline. 

I read the post on building a minimal docker container, and I agree with you that it would be more desirable. Let me know how I can help, I would encourage you to submit a pull request, and take it from there. You can probably start with the minimal Python environment mentioned in that post and use the current dockerfile (you can also refer to the `requirements.txt` and `python3.6-environment.yml` in the `etc` folder as a reference for the dependencies).

**Comment by dilzeem** [comment]
Hi, 

I was having very strange issues with this on a Windows 10 machine. 

When I try to git clone the repository and then build and run the container I would get an error saying. `/docker_cmd.sh: not found`.

I then proceed to entire the docker container to see if the indeed the file was copied across. It was but it seems like the shebang is not recognized in the file. 

I thought it was a line ending issue when I used Unix files in Windows but my Git does automatic conversion of the files. I Ahh ... I think I just figured it out. Git shouldn't be automatically converting my cloned files, because these are being cloned into a Linux system. 

Anyway for Windows Users wanting to use this repository just download the repository. I got it working that way.
